src/tests/samples/violations/conditional_prefer_type_error.py

```diff
@@ -0,0 +1,258 @@
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_and_operator_isinstance(arg1, arg2):
+    if isinstance(some_arg) and isinstance(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_or_operator_isinstance(arg1, arg2):
+    if isinstance(some_arg) or isinstance(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_multiple_operators_isinstance(arg1, arg2, arg3):
+    if not isinstance(arg1) and isinstance(arg2) or isinstance(arg3):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_not_operator_callable(some_arg):
+    if not callable(some_arg):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_and_operator_callable(arg1, arg2):
+    if callable(some_arg) and callable(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_or_operator_callable(arg1, arg2):
+    if callable(some_arg) or callable(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_multiple_operators_callable(arg1, arg2, arg3):
+    if not callable(arg1) and callable(arg2) or callable(arg3):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_not_operator_issubclass(some_arg):
+    if not issubclass(some_arg):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_and_operator_issubclass(arg1, arg2):
+    if issubclass(some_arg) and issubclass(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_or_operator_issubclass(arg1, arg2):
+    if issubclass(some_arg) or issubclass(arg2):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_multiple_operators_issubclass(arg1, arg2, arg3):
+    if not issubclass(arg1) and issubclass(arg2) or issubclass(arg3):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_multi_conditional(arg1, arg2):
+    if isinstance(arg1, int):
+        pass
+    elif isinstance(arg2, int):
+        raise Exception("...")  # should be typeerror
+
+
+class MyCustomTypeValidation(Exception):
+    pass
+
+
+def correct_custom_exception(some_arg):
+    if isinstance(some_arg, int):
+        pass
+    else:
+        raise MyCustomTypeValidation("...")  # that's correct, because it's not vanilla
+
+
+def correct_complex_conditional(val):
+    if val is not None and (not isinstance(val, int) or val < 0):
+        raise ValueError(...)  # fine if this is not a TypeError
+
+
+def correct_multi_conditional(some_arg):
+    if some_arg == 3:
+        pass
+    elif isinstance(some_arg, int):
+        pass
+    else:
+        raise Exception("...")  # fine if this is not a TypeError
```

Comment:
This case should **not** trigger a violation because not all conditions check for a type